plugins/plugin-maven/che-plugin-maven-generator-archetype/src/main/java/org/eclipse/che/plugin/maven/generator/archetype/ArchetypeGenerator.java

```diff
@@ -60,394 +45,109 @@
+                throw new IOException("Process failed. Exit code " + process.exitValue() + " command-line : " + Arrays.toString(commandLine));
             }
-        } catch (IOException e) {
-            throw new ServerException(e);
         } finally {
-            if (logReader != null) {
-                try {
-                    logReader.close();
-                } catch (IOException ignored) {
-                }
-            }
+            watcher.stop();
         }
-
-        if (!mavenSuccess) {
-            return new GenerationResult(false, null, getLogFile(task));
-        }
-
-        final File workDir = task.getWorkDir();
-        final GenerationResult result = new GenerationResult(true, null, getLogFile(task));
-
-        final File projectFolder = new File(workDir, task.getArtifactId());
-        if (projectFolder.isDirectory() && projectFolder.list().length > 0) {
-            final File zip = new File(workDir, "project.zip");
-            try {
-                ZipUtils.zipDir(projectFolder.getAbsolutePath(), projectFolder, zip, IoUtil.ANY_FILTER);
-            } catch (IOException e) {
-                throw new ServerException(e);
-            }
-            result.setGeneratedProject(zip);
-        }
-
-        return result;
     }
 
-    private File getLogFile(GenerationTask task) {
-        return task.getLogger().getFile();
-    }
-
-    private void cleanup(GenerationTask task) {
-        File workDir = task.getWorkDir();
-        if (workDir != null && workDir.exists()) {
-            if (!IoUtil.deleteRecursive(workDir)) {
-                LOG.warn("Unable to delete directory {}", workDir);
-            }
-        }
-    }
-
-    /** Logger that will write to file all the logs of the project generation process. */
-    private static class GeneratorLogger implements LineConsumer {
-        private final File    file;
-        private final Writer  writer;
-        private final boolean autoFlush;
-
-        GeneratorLogger(File file) throws IOException {
-            this.file = file;
-            autoFlush = true;
-            writer = Files.newBufferedWriter(file.toPath(), Charset.defaultCharset());
-        }
-
-        Reader getReader() throws IOException {
-            return Files.newBufferedReader(file.toPath(), Charset.defaultCharset());
-        }
-
-        /** Get {@code File} where logs stored. */
-        File getFile() {
-            return file;
-        }
-
-        @Override
-        public void writeLine(String line) throws IOException {
-            if (line != null) {
-                writer.write(line);
-            }
-            writer.write('\n');
-            if (autoFlush) {
-                writer.flush();
-            }
-        }
-
-        @Override
-        public void close() throws IOException {
-            writer.close();
-        }
+    /**
+     * Create specified command
```

Comment:
What command? Specified by what? Missed description of `@return` value.
IMO there's no need for javadoc for this private method.
WDYT?